instructorStudentComment: Give more informative status messages

_From [dam...@gmail.com](https://code.google.com/u/114919422028123491955/) on January 04, 2014 11:26:04_

Instead of generic messages, be more informative. e.g. The comment was deleted, also give an extract of the comment "The comment "blah blah" was deleted".

_Original issue: http://code.google.com/p/teammatespes/issues/detail?id=1427_

w.r.t. #6358 
Does anyone have an idea why this is happening?
> java.lang.AssertionError: expected:<New comment 'this is a comment' has been added> but was:<New comment '
> this is a comment
> ' has been added>

I've tried using a .trim() on the comment before passing it into String.format() but this is still happening.

--- a/src/main/java/teammates/common/util/Const.java
+++ b/src/main/java/teammates/common/util/Const.java
@@ -1367,7 +1367,7 @@ public final class Const {
                             + " to search for an exact phrase in an exact order.</li>"
                 + "</ul>";
         
-        public static final String COMMENT_ADDED = "New comment has been added";
+        public static final String COMMENT_ADDED = "New comment '%s' has been added";
         public static final String COMMENT_EDITED = "Comment edited";
         public static final String COMMENT_DELETED = "Comment deleted";
         public static final String COMMENT_CLEARED = "Notification for all pending comments have been sent to recipients";

--- a/src/main/java/teammates/common/util/Const.java
+++ b/src/main/java/teammates/common/util/Const.java
@@ -1367,7 +1367,7 @@ public final class Const {
                             + " to search for an exact phrase in an exact order.</li>"
                 + "</ul>";
         
-        public static final String COMMENT_ADDED = "New comment '%s' has been added";
+        public static final String COMMENT_ADDED = "New comment %s has been added";
         public static final String COMMENT_EDITED = "Comment edited";
         public static final String COMMENT_DELETED = "Comment deleted";
         public static final String COMMENT_CLEARED = "Notification for all pending comments have been sent to recipients";

--- a/src/main/java/teammates/common/util/Const.java
+++ b/src/main/java/teammates/common/util/Const.java
@@ -1367,7 +1367,7 @@ public final class Const {
                             + " to search for an exact phrase in an exact order.</li>"
                 + "</ul>";
         
-        public static final String COMMENT_ADDED = "New comment %s has been added";
+        public static final String COMMENT_ADDED = "Added new comment: %s";
         public static final String COMMENT_EDITED = "Comment edited";
         public static final String COMMENT_DELETED = "Comment deleted";
         public static final String COMMENT_CLEARED = "Notification for all pending comments have been sent to recipients";

--- a/src/main/java/teammates/ui/controller/InstructorStudentCommentAddAction.java
+++ b/src/main/java/teammates/ui/controller/InstructorStudentCommentAddAction.java
@@ -61,7 +61,9 @@ public class InstructorStudentCommentAddAction extends Action {
             //TODO: move putDocument to Task Queue
             logic.putDocument(createdComment);
             
-            statusToUser.add(new StatusMessage(Const.StatusMessages.COMMENT_ADDED, StatusMessageColor.SUCCESS));
+            statusToUser.add(new StatusMessage(String.format(Const.StatusMessages.COMMENT_ADDED,
+                                                             createdComment.getCommentText().trim()),
+                                               StatusMessageColor.SUCCESS));
             statusToAdmin = "Created Comment for Student:<span class=\"bold\">("
                             + comment.recipients + ")</span> for Course <span class=\"bold\">["
                             + comment.courseId + "]</span><br>"

--- a/src/main/java/teammates/ui/controller/InstructorStudentCommentAddAction.java
+++ b/src/main/java/teammates/ui/controller/InstructorStudentCommentAddAction.java
@@ -5,6 +5,9 @@ import java.util.Date;
 import java.util.HashSet;
 import java.util.List;
 
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
 import teammates.common.datatransfer.CommentAttributes;
 import teammates.common.datatransfer.CommentParticipantType;
 import teammates.common.datatransfer.CommentSendingState;
@@ -59,9 +62,8 @@ public class InstructorStudentCommentAddAction extends Action {
         try {
             CommentAttributes createdComment = logic.createComment(comment);
             //TODO: move putDocument to Task Queue
-
-            statusToUser.add(new StatusMessage(String.format(Const.StatusMessages.COMMENT_ADDED,
-                                                             createdComment.getCommentText()),
+            String commentPlainText = Jsoup.clean(createdComment.getCommentText(), Whitelist.none());
+            statusToUser.add(new StatusMessage(String.format(Const.StatusMessages.COMMENT_ADDED, commentPlainText),
                                                StatusMessageColor.SUCCESS));
             statusToAdmin = "Created Comment for Student:<span class=\"bold\">("
                             + comment.recipients + ")</span> for Course <span class=\"bold\">["

--- a/src/main/java/teammates/ui/controller/InstructorStudentCommentAddAction.java
+++ b/src/main/java/teammates/ui/controller/InstructorStudentCommentAddAction.java
@@ -59,10 +59,9 @@ public class InstructorStudentCommentAddAction extends Action {
         try {
             CommentAttributes createdComment = logic.createComment(comment);
             //TODO: move putDocument to Task Queue
-            logic.putDocument(createdComment);
-            
+
             statusToUser.add(new StatusMessage(String.format(Const.StatusMessages.COMMENT_ADDED,
-                                                             createdComment.getCommentText().trim()),
+                                                             createdComment.getCommentText()),
                                                StatusMessageColor.SUCCESS));
             statusToAdmin = "Created Comment for Student:<span class=\"bold\">("
                             + comment.recipients + ")</span> for Course <span class=\"bold\">["

--- a/src/main/java/teammates/ui/controller/InstructorStudentCommentAddAction.java
+++ b/src/main/java/teammates/ui/controller/InstructorStudentCommentAddAction.java
@@ -62,6 +62,7 @@ public class InstructorStudentCommentAddAction extends Action {
         try {
             CommentAttributes createdComment = logic.createComment(comment);
             //TODO: move putDocument to Task Queue
+            logic.putDocument(createdComment);
             String commentPlainText = Jsoup.clean(createdComment.getCommentText(), Whitelist.none());
             statusToUser.add(new StatusMessage(String.format(Const.StatusMessages.COMMENT_ADDED, commentPlainText),
                                                StatusMessageColor.SUCCESS));

--- a/src/test/java/teammates/test/cases/ui/InstructorStudentCommentAddActionTest.java
+++ b/src/test/java/teammates/test/cases/ui/InstructorStudentCommentAddActionTest.java
@@ -139,7 +139,8 @@ public class InstructorStudentCommentAddActionTest extends BaseActionTest {
                 + "error=false",
                 r.getDestinationWithParams());
         assertFalse(r.isError);
-        assertEquals("New comment has been added", r.getStatusMessage());
+        assertEquals(String.format(Const.StatusMessages.COMMENT_ADDED, submissionParams[1]),
+                     r.getStatusMessage());
 
         String expectedLogMessage =
                 "TEAMMATESLOG|||instructorStudentCommentAdd|||instructorStudentCommentAdd"

--- a/src/test/java/teammates/test/cases/ui/InstructorStudentCommentAddActionTest.java
+++ b/src/test/java/teammates/test/cases/ui/InstructorStudentCommentAddActionTest.java
@@ -173,7 +173,8 @@ public class InstructorStudentCommentAddActionTest extends BaseActionTest {
                 + "error=false",
                 r.getDestinationWithParams());
         assertFalse(r.isError);
-        assertEquals("New comment has been added", r.getStatusMessage());
+        assertEquals(String.format(Const.StatusMessages.COMMENT_ADDED, submissionParams[1]),
+                     r.getStatusMessage());
 
         expectedLogMessage =
                 "TEAMMATESLOG|||instructorStudentCommentAdd|||instructorStudentCommentAdd"
@@ -207,7 +208,8 @@ public class InstructorStudentCommentAddActionTest extends BaseActionTest {
                 + "error=false",
                 r.getDestinationWithParams());
         assertFalse(r.isError);
-        assertEquals("New comment has been added", r.getStatusMessage());
+        assertEquals(String.format(Const.StatusMessages.COMMENT_ADDED, submissionParams[1]),
+                     r.getStatusMessage());
 
         expectedLogMessage =
                 "TEAMMATESLOG|||instructorStudentCommentAdd|||instructorStudentCommentAdd"
@@ -240,7 +242,8 @@ public class InstructorStudentCommentAddActionTest extends BaseActionTest {
                 + "error=false",
                 r.getDestinationWithParams());
         assertFalse(r.isError);
-        assertEquals("New comment has been added", r.getStatusMessage());
+        assertEquals(String.format(Const.StatusMessages.COMMENT_ADDED, submissionParams[1]),
+                     r.getStatusMessage());
 
         expectedLogMessage =
                 "TEAMMATESLOG|||instructorStudentCommentAdd|||instructorStudentCommentAdd"
@@ -273,7 +276,8 @@ public class InstructorStudentCommentAddActionTest extends BaseActionTest {
                 + "error=false",
                 r.getDestinationWithParams());
         assertFalse(r.isError);
-        assertEquals("New comment has been added", r.getStatusMessage());
+        assertEquals(String.format(Const.StatusMessages.COMMENT_ADDED, submissionParams[1]),
+                     r.getStatusMessage());
 
         expectedLogMessage =
                 "TEAMMATESLOG|||instructorStudentCommentAdd|||instructorStudentCommentAdd"
@@ -306,7 +310,8 @@ public class InstructorStudentCommentAddActionTest extends BaseActionTest {
                 + "error=false",
                 r.getDestinationWithParams());
         assertFalse(r.isError);
-        assertEquals("New comment has been added", r.getStatusMessage());
+        assertEquals(String.format(Const.StatusMessages.COMMENT_ADDED, submissionParams[1]),
+                     r.getStatusMessage());
 
         expectedLogMessage =
                 "TEAMMATESLOG|||instructorStudentCommentAdd|||instructorStudentCommentAdd"
@@ -339,7 +344,8 @@ public class InstructorStudentCommentAddActionTest extends BaseActionTest {
                 + "error=false",
                 r.getDestinationWithParams());
         assertFalse(r.isError);
-        assertEquals("New comment has been added", r.getStatusMessage());
+        assertEquals(String.format(Const.StatusMessages.COMMENT_ADDED, submissionParams[1]),
+                     r.getStatusMessage());
 
         expectedLogMessage =
                 "TEAMMATESLOG|||instructorStudentCommentAdd|||instructorStudentCommentAdd"
@@ -372,7 +378,8 @@ public class InstructorStudentCommentAddActionTest extends BaseActionTest {
                 + "error=false",
                 r.getDestinationWithParams());
         assertFalse(r.isError);
-        assertEquals("New comment has been added", r.getStatusMessage());
+        assertEquals(String.format(Const.StatusMessages.COMMENT_ADDED, submissionParams[1]),
+                     r.getStatusMessage());
 
         expectedLogMessage =
                 "TEAMMATESLOG|||instructorStudentCommentAdd|||instructorStudentCommentAdd"
@@ -406,7 +413,8 @@ public class InstructorStudentCommentAddActionTest extends BaseActionTest {
                 + "error=false",
                 r.getDestinationWithParams());
         assertFalse(r.isError);
-        assertEquals("New comment has been added", r.getStatusMessage());
+        assertEquals(String.format(Const.StatusMessages.COMMENT_ADDED, submissionParams[1]),
+                     r.getStatusMessage());
 
         expectedLogMessage =
                 "TEAMMATESLOG|||instructorStudentCommentAdd|||instructorStudentCommentAdd"
@@ -440,7 +448,8 @@ public class InstructorStudentCommentAddActionTest extends BaseActionTest {
                 + "error=false",
                 r.getDestinationWithParams());
         assertFalse(r.isError);
-        assertEquals("New comment has been added", r.getStatusMessage());
+        assertEquals(String.format(Const.StatusMessages.COMMENT_ADDED, submissionParams[1]),
+                     r.getStatusMessage());
 
         expectedLogMessage =
                 "TEAMMATESLOG|||instructorStudentCommentAdd|||instructorStudentCommentAdd"

--- a/src/test/java/teammates/test/cases/ui/browsertests/InstructorCourseDetailsPageUiTest.java
+++ b/src/test/java/teammates/test/cases/ui/browsertests/InstructorCourseDetailsPageUiTest.java
@@ -110,8 +110,9 @@ public class InstructorCourseDetailsPageUiTest extends BaseUiTestCase {
         detailsPage.clickAddCommentToCourseButton();
         
         ______TS("comment to whole course: any comment");
-        detailsPage.submitCommentToCourse("this is a comment");
-        detailsPage.verifyStatus(Const.StatusMessages.COMMENT_ADDED);
+        String commentText = "this is a comment";
+        detailsPage.submitCommentToCourse(commentText);
+        detailsPage.verifyStatus(String.format(Const.StatusMessages.COMMENT_ADDED, commentText));
     }
 
     private void testTableSort() {

--- a/src/test/java/teammates/test/cases/ui/browsertests/InstructorStudentRecordsPageUiTest.java
+++ b/src/test/java/teammates/test/cases/ui/browsertests/InstructorStudentRecordsPageUiTest.java
@@ -179,7 +179,8 @@ public class InstructorStudentRecordsPageUiTest extends BaseUiTestCase {
         ______TS("add comment: success");
 
         viewPage.addComment("New comment from teammates.test for Benny C")
-                .verifyStatus("New comment has been added");
+                .verifyStatus(String.format(Const.StatusMessages.COMMENT_ADDED,
+                                            "New comment from teammates.test for Benny C"));
 
         ______TS("add comment with custom visibility: success");
         

--- a/src/test/resources/pages/instructorStudentRecordsPageAddComment.html
+++ b/src/test/resources/pages/instructorStudentRecordsPageAddComment.html
@@ -10,7 +10,7 @@
   <br>
   <div id="statusMessagesToUser">
     <div class="overflow-auto alert alert-success statusMessage">
-      New comment has been added
+      Added new comment: New comment from teammates.test for Benny C, viewable by everyone
     </div>
   </div>
   <script defer="" src="/js/statusMessage.js" type="text/javascript">

